### PR TITLE
txt2tags for python3 #8734

### PR DESCRIPTION
**Pull request by nishimotz**
### Link to issue number:

#8734

### Summary of the issue:

This PR is against the threshold branch.
It replaces txt2tags.py and keyCommandDoc.py with my work of https://github.com/nvdajp/txt2tags.
Note that nvda-misc-deps is forked and https://github.com/nvdajp/nvda-misc-deps is used.

### Description of how this pull request fixes the issue:

Part of this repository is verified that it can make the same documents using Python 3.7.3.

### Testing performed:

This repository itself is not tested.

### Change log entry:

Section: New features, Changes, Bug fixes

### CODE REVIEW

**Comment by zstanecic** [comment]
Hi,

I am against putting the changelog item for this into the new features section, at it is a developers change not visible for the end-user.

File: keyCommandsDoc.py [hunk 1]
```diff
@@ -13,7 +14,7 @@
 import re
 import txt2tags
 
-LINE_END = u"\r\n"
+LINE_END = "\r\n"
```
Comment (leonardder) [inline keyCommandsDoc.py]: I'm not against this change, but it is unnecessary.

File: keyCommandsDoc.py [hunk 2]
```diff
@@ -195,7 +196,7 @@ def _writeHeadings(self):
 		for level, heading in enumerate(self._headings[level:], level):
 			# We don't want numbered headings in the output.
 			label=heading.group("label")
-			headingText = u"{id}{txt}{id}{label}".format(
+			headingText = "{id}{txt}{id}{label}".format(
```
Comment (leonardder) [inline keyCommandsDoc.py]: Same as above

File: keyCommandsDoc.py [hunk 3]
```diff
@@ -266,9 +267,9 @@ def _handleSetting(self):
 		desc = next(self._ug).strip()
 		self._lineNum += 1
 
-		self._kc.write(u"| {name} | {keys} | {desc} |{lineEnd}".format(
+		self._kc.write("| {name} | {keys} | {desc} |{lineEnd}".format(
```
Comment (leonardder) [inline keyCommandsDoc.py]: Same as above

File: keyCommandsDoc.py [hunk 4]
```diff
@@ -266,9 +267,9 @@ def _handleSetting(self):
 		desc = next(self._ug).strip()
 		self._lineNum += 1
 
-		self._kc.write(u"| {name} | {keys} | {desc} |{lineEnd}".format(
+		self._kc.write("| {name} | {keys} | {desc} |{lineEnd}".format(
 			name=name,
-			keys=u" | ".join(keys),
+			keys=" | ".join(keys),
```
Comment (leonardder) [inline keyCommandsDoc.py]: Same as above

**Review by LeonarddeR — CHANGES_REQUESTED** [review]
1. Could you also process site_scons/site_tools/t2t.py
2. It would be good to know what you did to get this result. Was it simply running the code through 2to3?
3. I assume you're still planning to file a pr against nvda-misc-deps for txt2tags itself?

File: .gitmodules [hunk 5]
```diff
@@ -6,7 +6,7 @@
 	url = https://github.com/RaMMicHaeL/minhook
 [submodule "miscDeps"]
 	path = miscDeps
-	url = https://github.com/nvaccess/nvda-misc-deps
+	url = https://github.com/nvdajp/nvda-misc-deps
```
[the remark this answers is not in the document]
Comment (nishimotz) [reply .gitmodules]: 1. The threshold branch already has t2t.py which is the same as my work.

2. I have used modernizer, then manually tweaked regarding string.join and others. Finally, the same document is generated.

3. PR against nvda-misc-deps is as follows:
https://github.com/nvaccess/nvda-misc-deps/pull/12

File: keyCommandsDoc.py [hunk 6]
```diff
@@ -3,7 +3,8 @@
 #A part of NonVisual Desktop Access (NVDA)
 #This file is covered by the GNU General Public License.
 #See the file COPYING for more details.
-#Copyright (C) 2010-2015 NV Access Limited, Mesar Hameed
+#Copyright (C) 2010-2019 NV Access Limited, Mesar Hameed, Takuya Nishimoto
+#Copyright (C) 2018-2019 Takuya Nishimoto
```
Comment (leonardder) [inline keyCommandsDoc.py]: Please remove this line

**Review by LeonarddeR — CHANGES_REQUESTED** [review]
Just one thing

**Review by LeonarddeR — APPROVED** [review]

**Comment by leonardder** [comment]
@michaeldcurran: as soon as https://github.com/nvaccess/nvda-misc-deps/pull/12 is merged, this needs an update of misc-deps to master.

**Review by michaelDCurran — APPROVED** [review]

**Pull request merged**